src/main/java/com/github/lipiridi/spotless/applier/config/SpotlessOnSaveActionInfoBase.java

```diff
@@ -0,0 +1,252 @@
+// Copyright 2000-2021 JetBrains s.r.o. and contributors. Use of this source code is governed by the
+// Apache 2.0 license that can be found in the LICENSE file.
+
+package com.github.lipiridi.spotless.applier.config;
+
+import com.intellij.codeInsight.CodeInsightBundle;
+import com.intellij.ide.actionsOnSave.ActionOnSaveComment;
+import com.intellij.ide.actionsOnSave.ActionOnSaveContext;
+import com.intellij.ide.actionsOnSave.ActionOnSaveInfo;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeRegistry;
+import com.intellij.openapi.ui.popup.JBPopup;
+import com.intellij.openapi.ui.popup.JBPopupFactory;
+import com.intellij.openapi.ui.popup.JBPopupListener;
+import com.intellij.openapi.ui.popup.LightweightWindowEvent;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.NlsContexts;
+import com.intellij.ui.CheckboxTree;
+import com.intellij.ui.CheckedTreeNode;
+import com.intellij.ui.TreeSpeedSearch;
+import com.intellij.ui.components.DropDownLink;
+import com.intellij.ui.components.JBScrollPane;
+import java.util.*;
+import javax.swing.*;
+import javax.swing.tree.TreeNode;
+import kotlin.jvm.functions.Function1;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public abstract class SpotlessOnSaveActionInfoBase<Options extends SpotlessOnSaveOptionsBase<?>>
```

Comment:
May be not needed if SpotlessOnSaveActionInfo will extend FormatOnSaveActionInfoBase

Reply:
I don't think this would work, please re-check.